vmb/matrimony/models/profiles.py

```diff
@@ -1087,3 +1056,42 @@ class MatrimonyProfileStats(BaseModel):
 
     class Meta:
         db_table = "matrimony_profile_stats"
+
+
+def update_stats(sender, instance, **kwargs):
+    _, created = MatrimonyProfileStats.objects.get_or_create(profile=instance.male)
+    _, created = MatrimonyProfileStats.objects.get_or_create(profile=instance.female)
+
+    matches_suggested = (
+        matches_accepted
+    ) = matches_rejected = matches_accepted_by = matches_rejected_by = 0
+
+    for m in instance.male.matches:
+        matches_suggested += 1
+        matches_accepted += 1 if getattr(m, "male_response") == "ACP" else 0
+        matches_rejected += 1 if getattr(m, "male_response") == "REJ" else 0
+        matches_accepted_by += 1 if getattr(m, "female_response") == "ACP" else 0
+        matches_rejected_by += 1 if getattr(m, "female_response") == "REJ" else 0
+    instance.male.stats.matches_suggested = matches_suggested
+    instance.male.stats.matches_accepted = matches_accepted
+    instance.male.stats.matches_rejected = matches_rejected
+    instance.male.stats.matches_accepted_by = matches_accepted_by
+    instance.male.stats.matches_rejected_by = matches_rejected_by
+    instance.male.stats.save()
+
+    for m in instance.female.matches:
+        matches_suggested += 1
+        matches_accepted += 1 if getattr(m, "female_response") == "ACP" else 0
+        matches_rejected += 1 if getattr(m, "female_response") == "REJ" else 0
+        matches_accepted_by += 1 if getattr(m, "male_response") == "ACP" else 0
+        matches_rejected_by += 1 if getattr(m, "male_response") == "REJ" else 0
+    instance.female.stats.matches_suggested = matches_suggested
+    instance.female.stats.matches_accepted = matches_accepted
+    instance.female.stats.matches_rejected = matches_rejected
+    instance.female.stats.matches_accepted_by = matches_accepted_by
+    instance.female.stats.matches_rejected_by = matches_rejected_by
+    instance.female.stats.save()
+
+
+post_save.connect(update_stats, sender=Match)
+post_delete.connect(update_stats, sender=Match)
```

Comment:
This should go to `signals.py`.

Reply:
@bhawanabadlani any update on this?

Reply:
![image](https://user-images.githubusercontent.com/47107719/104413684-c72bf680-5594-11eb-9cc1-511164911ceb.png)
Amongst the three ways, which one should we use to call the class function?
Any one of the three can serve the purpose.
https://stackoverflow.com/questions/12549977/how-can-i-have-a-django-signal-call-a-model-method